lib/proxy/plugin.rb

```diff
@@ -116,10 +112,32 @@ def plugin(plugin_name, aversion)
     end
 
     def uses_provider
-      @uses_provider = true
+      @get_uses_provider = true
+    end
+  end
+
+  def validate_prerequisites_enabled!(all_plugins, prerequisites)
+    prerequisites.each do |p|
+      if !(all_plugins.find {|plugin| plugin[:name] == p.to_sym})
+        raise ::Proxy::PluginMisconfigured, "Unable to find dependency '#{p}' of '#{plugin_name}'."
+      end
+      if !(all_plugins.find {|plugin| plugin[:name] == p.to_sym && plugin[:enabled] == true})
+        raise ::Proxy::PluginMisconfigured, "Dependency '#{p}' of '#{plugin_name}' has not been enabled."
+      end
     end
   end
 
+  def validate!(all_plugins)
+    validate_dependencies!(all_plugins, self.class.dependencies)
+    validate_prerequisites_enabled!(all_plugins, [settings.use_provider]) if uses_provider?
```

Comment:
This method could probably stay in `Pluggable` as 3/4 of it's the same as the provider, and then call `validate_prerequisites_enabled!` & `super`.